### PR TITLE
Interactions Fixes

### PR DESCRIPTION
**Pull request by HarryET**
Changed HTTP Handler to view permission requests in a separate bucket (required dependency override till new nyxx version pushed). Fixed permission issues.

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

Use smart commits here to manipulate issues (eg. Fixes #issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [ ] Ran `dartanalyzer --options analysis_options.yaml .`
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: nyxx_interactions/lib/src/Interactions.dart [hunk 1]
```diff
@@ -85,11 +85,13 @@ class Interactions {
         _groupSlashCommandBuilders(commandPartition.last);
 
     final globalBody = globalCommands
-      .where((builder) => builder.permissions.isNotEmpty)
+      .where((builder) => builder.permissions != null)
+      .where((builder) => builder.permissions!.isNotEmpty)
```
Comment (l7ssha) [inline nyxx_interactions/lib/src/Interactions.dart:89]: should be combined together to avoit multiple iteration over colleciton

File: nyxx_interactions/lib/src/Interactions.dart [hunk 2]
```diff
@@ -98,11 +100,13 @@ class Interactions {
 
     for (final entry in groupedGuildCommands.entries) {
       final guildBody = entry.value
-        .where((builder) => builder.permissions.isNotEmpty)
+        .where((builder) => builder.permissions != null)
+        .where((builder) => builder.permissions!.isNotEmpty)
```
Comment (l7ssha) [inline nyxx_interactions/lib/src/Interactions.dart:104]: same as above

**Review by l7ssha — CHANGES_REQUESTED** [review]

File: nyxx_interactions/lib/src/Interactions.dart (hunk 1 above)
Comment (HarryET) [reply nyxx_interactions/lib/src/Interactions.dart]: fixed

File: nyxx_interactions/lib/src/Interactions.dart (hunk 2 above)
Comment (HarryET) [reply nyxx_interactions/lib/src/Interactions.dart]: fixed

**Pull request merged**